programs/client/Client.cpp

```diff
@@ -481,48 +482,84 @@ catch (...)
 
 void Client::connect()
 {
-    connection_parameters = ConnectionParameters(config());
-
-    if (is_interactive)
-        std::cout << "Connecting to "
-                    << (!connection_parameters.default_database.empty() ? "database " + connection_parameters.default_database + " at "
-                                                                        : "")
-                    << connection_parameters.host << ":" << connection_parameters.port
-                    << (!connection_parameters.user.empty() ? " as user " + connection_parameters.user : "") << "." << std::endl;
+    for (size_t attempted_address_index = 0; attempted_address_index < hosts_ports.size(); ++attempted_address_index)
+    {
+        DB::DNSResolver::instance().resolveHost(hosts_ports[attempted_address_index].host);
+    }
+    bool is_secure = config().getBool("secure", false);
+    String tcp_port_config_key = is_secure ? "tcp_port_secure" : "tcp_port";
+    UInt16 default_port = config().getInt("port",
+        config().getInt(tcp_port_config_key,
+            is_secure ? DBMS_DEFAULT_SECURE_PORT : DBMS_DEFAULT_PORT));
```

Comment:
Minor comment, I see the same code in ConnectionParameters.cpp lines 73-76, maybe it worth creating a small function that accepts config and return port from it

Reply:
Done